modules/core/shared/src/main/scala/gem/syntax/Stream.scala

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.syntax
+
+import fs2.{ Chunk, Pull, Stream }
+
+final class StreamOps[F[_], O](val self: Stream[F, O]) {
+
+  /** A filter/fold combination that uses the accumulated result of folding
+    * over the elements since the last element was emitted to determine whether
+    * the current element under examination should be emitted.  The filter
```

Comment:
Ok, this is admittedly a bit whacky.  I could be easily convinced that it doesn't really belong here and should be relegated to the ephemeris package where it is actually used.

Reply:
Looks good. I think you should PR this back to fs2 since it's completely generic, and then we can then delete it from our codebase.